HeliPort/Appearance/StatusMenu.swift

```diff
@@ -607,10 +616,13 @@ final class StatusMenu: NSMenu, NSMenuDelegate {
         let ssid = String(sender.title).replacingOccurrences(of: String.disconnectNet, with: "",
                                                              options: .regularExpression,
                                                              range: nil)
+        isDisconnecting = true
         DispatchQueue.global().async {
-            CredentialsManager.instance.setAutoJoin(ssid, false)
```

Comment:
Please correct me if I'm wrong, I did recall this is what Apple's implementation used to do back in the days - manually disconnecting a network will opt-out from auto-join, although this is not what the latest macOS versions are doing.

Reply:
You are correct. Apple’s implementation used to work this way: manually disconnecting from a network would remove it from auto-join. However, more recent versions of macOS no longer follow this behavior; Apple made this change to improve user experience by reducing the need to manually reconfigure network preferences.

Reply:
Would you prefer to keep this behavior of automatically disabling autojoin when the user disconnects? I think this behavior is not very interesting and many users may not like this behavior. If you prefer to keep this behavior, I think it would be better to add an option in the preferences to activate and deactivate this behavior.